Core/MockInstantiationRewriter.cs

```diff
@@ -0,0 +1,128 @@
+//
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace RhinoMocksToMoqRewriter.Core
+{
+  public class MockInstantiationRewriter : CSharpSyntaxRewriter
+  {
+    private readonly SemanticModel _model;
+    private const string c_generateStrictMock = "GenerateStrictMock";
+    private const string c_indent = "        ";
+
+    public MockInstantiationRewriter (SemanticModel model)
+    {
+      _model = model;
+    }
+
+    public override SyntaxNode? VisitInvocationExpression (InvocationExpressionSyntax node)
+    {
+      var compilationSymbol = _model.Compilation.GetTypeByMetadataName ("Rhino.Mocks.MockRepository");
+      if (compilationSymbol == null)
+      {
+        throw new ArgumentException ("Rhino.Mocks cannot be found.");
+      }
+
+      var methodSymbol = _model.GetSymbolInfo (node).Symbol as IMethodSymbol;
+      if (methodSymbol == null)
+      {
+        return node;
+      }
+
+      var isRhinoMocksMethod = SymbolEqualityComparer.Default.Equals (methodSymbol.ContainingType, compilationSymbol);
+      if (!isRhinoMocksMethod)
+      {
+        return node;
+      }
+
+      var rhinoMethodGenericName = node.DescendantNodes().FirstOrDefault (n => n.IsKind (SyntaxKind.GenericName)) as GenericNameSyntax;
+      if (rhinoMethodGenericName == null || rhinoMethodGenericName.Identifier.Text != methodSymbol.Name)
+      {
+        return node;
+      }
+
+      var moqMockTypeArgumentList = SyntaxFactory.TypeArgumentList().AddArguments (rhinoMethodGenericName.TypeArgumentList.Arguments.First());
+      var moqMockArgumentSyntaxList = SyntaxFactory.SeparatedList<ArgumentSyntax>().AddRange (node.ArgumentList.Arguments);
+
+      if (methodSymbol.Name == c_generateStrictMock)
+      {
+        var strictArgument =
+            SyntaxFactory.Argument (
+                SyntaxFactory.MemberAccessExpression (
+                        SyntaxKind.SimpleMemberAccessExpression,
+                        SyntaxFactory.IdentifierName ("MockBehaviour"),
+                        SyntaxFactory.IdentifierName ("Strict"))
+                    .WithOperatorToken (
+                        SyntaxFactory.Token (SyntaxKind.DotToken)));
+
+        moqMockArgumentSyntaxList = moqMockArgumentSyntaxList
+            .Insert (0, strictArgument);
+      }
+
+      var multiLineIndent = GetNewLineCharacter (node.ArgumentList);
+
+      for (var i = 0; i < moqMockArgumentSyntaxList.Count; i++)
+      {
+        var argument = moqMockArgumentSyntaxList[i];
+        var indent = argument.HasLeadingTrivia
+            ? argument.GetLeadingTrivia().First().ToString()
+            : SyntaxFactory.Space.ToString();
+
+        var firsLineIndent = string.IsNullOrEmpty (multiLineIndent)
+            ? string.Empty
+            : c_indent;
```

Comment:
The needed indentation should rather be calculated than fixed to 8 spaces. If the containing method is a local method e.g., the indentation would be wrong, same if all of this happens in a nested class or namespace, or when instantiating a field directly. Please try to get the leading trivia of the containing statement and add 4 spaces to get the indentation for the multiline arguments.

Reply:
Please also add more specific tests for these scenarios.